R/new_covariate.R

```diff
@@ -30,6 +30,7 @@ new_covariate <- function(
     times <- c(0)
     value <- value[1]
   }
+  implementation <- match.arg(tolower(implementation), c("interpolate", "locf"))
```

Comment:
Maybe we should make the default value for the `implementation` argument `c("interpolate", "locf")` to convey what the two options are, then this line could just be `implementation <- match.arg(implementation)`?

Reply:
I've updated as per your suggestion. Since we use a mix of 'locf' and 'LOCF' across documentation and testing I wanted to make sure we were not case-sensitive in our check so it was slightly more complicated.